libsplinter/src/circuit/template/rules/set_metadata.rs

```diff
@@ -12,13 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use super::super::yaml_parser::v1;
-use super::Value;
+use super::super::{yaml_parser::v1, CircuitTemplateError, CreateCircuitBuilder};
+use super::{get_argument_value, is_arg, RuleArgument, Value};
 
 pub(super) struct SetMetadata {
     metadata: Metadata,
 }
 
+impl SetMetadata {
+    pub fn apply_rule(
+        &self,
+        builder: CreateCircuitBuilder,
+        template_arguments: &[RuleArgument],
+    ) -> Result<CreateCircuitBuilder, CircuitTemplateError> {
+        match &self.metadata {
+            Metadata::Json { metadata } => {
+                let mut metadata = metadata
+                    .iter()
+                    .try_fold::<_, _, Result<_, CircuitTemplateError>>(
+                        "{[".to_string(),
```

Comment:
Would probably be easier to do this by just constructing a `Vec` in the `try_fold`, `collecting`ing, and then `join`ing (plus it would get rid of any trailing commas).

Reply:
Addressed